File: NonsensePoemGenerator.java
import java.util.Random;

/**
 * Nonsense Poem Generator - Java Edition
 * Generates random nonsensical poems using word banks and templates
 */
public class NonsensePoemGenerator {
    
    private static final String[] ADJECTIVES = {
        "purple", "bouncy", "invisible", "squeaky", "jellied", "grumpy",
        "wobbly", "sparkly", "fuzzy", "peculiar", "luminous", "dizzy",
        "twisted", "bubbly", "slippery", "crooked", "jumbled", "frazzled"
    };
    
    private static final String[] NOUNS = {
        "penguin", "toaster", "umbrella", "banana", "telescope", "accordion",
        "spatula", "moonbeam", "pickle", "flamingo", "waffle", "bicycle",
        "teapot", "mushroom", "harmonica", "walrus", "pretzel", "chandelier"
    };
    
    private static final String[] VERBS = {
        "dances", "wiggles", "somersaults", "yodels", "hiccups", "juggles",
        "sneezes", "bounces", "twirls", "prances", "waddles", "zigzags",
        "tumbles", "flutters", "wobbles", "giggles", "shuffles", "leaps"
    };
    
    private static final String[] ADVERBS = {
        "backwards", "sideways", "slowly", "frantically", "mysteriously", "gleefully",
        "awkwardly", "gracefully", "clumsily", "eagerly", "suddenly", "softly",
        "wildly", "gently", "rapidly", "lazily", "merrily", "oddly"
    };
    
    private static final String[] PREPOSITIONS = {
        "under", "over", "through", "around", "beneath", "beside",
        "above", "below", "between", "among", "beyond", "inside"
    };
    
    private static final String[] PLACES = {
        "the garden", "a cloud", "the meadow", "the ocean", "a rainbow",
        "the attic", "a dream", "the forest", "the stars", "a puddle",
        "the moon", "a teacup", "the shadows", "the wind", "a bubble"
    };
    
    private final Random random;
    
    public NonsensePoemGenerator() {
        this.random = new Random();
    }
    
    private String pick(String[] array) {
        return array[random.nextInt(array.length)];
    }
    
    private String capitalize(String str) {
        if (str == null || str.isEmpty()) {
            return str;
        }
        return Character.toUpperCase(str.charAt(0)) + str.substring(1);
    }
    
    private String generateTemplate1() {
        String adj1 = pick(ADJECTIVES);
        String noun1 = pick(NOUNS);
        String verb1 = pick(VERBS);
        String adv1 = pick(ADVERBS);
        String adj2 = pick(ADJECTIVES);
        String noun2 = pick(NOUNS);
        String prep1 = pick(PREPOSITIONS);
        String place1 = pick(PLACES);
        
        return String.format("The %s %s %s %s,\n" +
                           "While %s %ss rest %s %s.\n" +
                           "With %s delight and %s glee,\n" +
                           "They %s together wild and free!",
                           adj1, noun1, verb1, adv1,
                           adj2, noun2, prep1, place1,
                           pick(ADJECTIVES), pick(ADJECTIVES),
                           pick(VERBS));
    }
    
    private String generateTemplate2() {
        return String.format("%s %s\n%s %s %s %s\n%s %s %s",
                           pick(ADJECTIVES), pick(NOUNS),
                           pick(VERBS), pick(ADVERBS), pick(PREPOSITIONS), pick(PLACES),
                           pick(ADJECTIVES), pick(NOUNS), pick(VERBS));
    }
    
    private String generateTemplate3() {
        String noun1 = pick(NOUNS);
        String noun2 = pick(NOUNS);
        String noun3 = pick(NOUNS);
        
        return String.format("Once a %s %s met a %s %s,\n" +
                           "They %s %s %s %s.\n" +
                           "\"%s %s!\" the %s cried,\n" +
                           "As a %s %s %s by its side.",
                           pick(ADJECTIVES), noun1, pick(ADJECTIVES), noun2,
                           pick(VERBS), pick(ADVERBS), pick(PREPOSITIONS), pick(PLACES),
                           capitalize(pick(ADJECTIVES)), pick(NOUNS), noun1,
                           pick(ADJECTIVES), noun3, pick(VERBS));
    }
    
    private String generateTemplate4() {
        String verb = pick(VERBS);
        String noun = pick(NOUNS);
        String adj = pick(ADJECTIVES);
        String adv = pick(ADVERBS);
        
        return String.format("The %s %s,\n%s it %s,\n%s and %s it %s,\n" +
                           "Oh, how that %s %s!",
                           noun, verb, adv, verb, adj, pick(ADJECTIVES), verb,
                           noun, verb);
    }
    
    private String generateTemplate5() {
        String noun1 = pick(NOUNS);
        String noun2 = pick(NOUNS);
        
        return String.format("What happens when a %s %s\n" +
                           "Meets a %s %s %s %s?\n" +
                           "They %s and %s %s,\n" +
                           "Creating %s surprises!",
                           pick(ADJECTIVES), noun1,
                           pick(ADJECTIVES), noun2, pick(PREPOSITIONS), pick(PLACES),
                           pick(VERBS), pick(VERBS), pick(ADVERBS),
                           pick(ADJECTIVES));
    }
    
    private String generateTemplate6() {
        return String.format("I saw:\n" +
                           "A %s %s that %s %s,\n" +
                           "A %s %s %s %s,\n" +
                           "And a %s %s doing %s things,\n" +
                           "All while the %s %s %s!",
                           pick(ADJECTIVES), pick(NOUNS), pick(VERBS), pick(ADVERBS),
                           pick(ADJECTIVES), pick(NOUNS), pick(PREPOSITIONS), pick(PLACES),
                           pick(ADJECTIVES), pick(NOUNS), pick(ADJECTIVES),
                           pick(ADJECTIVES), pick(NOUNS), pick(VERBS));
    }
    
    public String generatePoem() {
        int template = random.nextInt(6);
        switch (template) {
            case 0: return generateTemplate1();
            case 1: return generateTemplate2();
            case 2: return generateTemplate3();
            case 3: return generateTemplate4();
            case 4: return generateTemplate5();
            case 5: return generateTemplate6();
            default: return generateTemplate1();
        }
    }
    
    public void generatePoems(int count) {
        System.out.println("=".repeat(60));
        System.out.println("  🎭 NONSENSE POEM GENERATOR (Java) 🎭");
        System.out.println("=".repeat(60));
        System.out.println();
        
        for (int i = 1; i <= count; i++) {
            System.out.println("Poem #" + i + ":");
            System.out.println("-".repeat(60));
            System.out.println(generatePoem());
            System.out.println();
            
            if (i < count) {
                System.out.println();
            }
        }
        
        System.out.println("=".repeat(60));
    }
    
    public static void main(String[] args) {
        int count = 3;
        
        if (args.length > 0) {
            try {
                count = Integer.parseInt(args[0]);
                if (count < 1) {
                    throw new IllegalArgumentException();
                }
            } catch (Exception e) {
                System.err.println("Usage: java NonsensePoemGenerator [number_of_poems]");
                System.err.println("Example: java NonsensePoemGenerator 5");
                System.exit(1);
            }
        }
        
        NonsensePoemGenerator generator = new NonsensePoemGenerator();
        generator.generatePoems(count);
    }
}

File: nonsense-poem-generator.cpp
#include <iostream>
#include <string>
#include <vector>
#include <random>
#include <functional>
#include <algorithm>

// Nonsense Poem Generator - C++ Edition

class PoemGenerator {
private:
    std::vector<std::string> adjectives = {
        "purple", "bouncy", "invisible", "squeaky", "jellied", "grumpy",
        "wobbly", "sparkly", "fuzzy", "peculiar", "luminous", "dizzy",
        "twisted", "bubbly", "slippery", "crooked", "jumbled", "frazzled"
    };
    
    std::vector<std::string> nouns = {
        "penguin", "toaster", "umbrella", "banana", "telescope", "accordion",
        "spatula", "moonbeam", "pickle", "flamingo", "waffle", "bicycle",
        "teapot", "mushroom", "harmonica", "walrus", "pretzel", "chandelier"
    };
    
    std::vector<std::string> verbs = {
        "dances", "wiggles", "somersaults", "yodels", "hiccups", "juggles",
        "sneezes", "bounces", "twirls", "prances", "waddles", "zigzags",
        "tumbles", "flutters", "wobbles", "giggles", "shuffles", "leaps"
    };
    
    std::vector<std::string> adverbs = {
        "backwards", "sideways", "slowly", "frantically", "mysteriously", "gleefully",
        "awkwardly", "gracefully", "clumsily", "eagerly", "suddenly", "softly",
        "wildly", "gently", "rapidly", "lazily", "merrily", "oddly"
    };
    
    std::vector<std::string> prepositions = {
        "under", "over", "through", "around", "beneath", "beside",
        "above", "below", "between", "among", "beyond", "inside"
    };
    
    std::vector<std::string> places = {
        "the garden", "a cloud", "the meadow", "the ocean", "a rainbow",
        "the attic", "a dream", "the forest", "the stars", "a puddle",
        "the moon", "a teacup", "the shadows", "the wind", "a bubble"
    };
    
    std::mt19937 rng;
    
    template<typename T>
    const T& pick(const std::vector<T>& vec) {
        std::uniform_int_distribution<size_t> dist(0, vec.size() - 1);
        return vec[dist(rng)];
    }
    
    std::string capitalize(std::string str) {
        if (!str.empty()) {
            str[0] = std::toupper(str[0]);
        }
        return str;
    }
    
    std::string generateTemplate1() {
        std::string result;
        auto adj1 = pick(adjectives);
        auto noun1 = pick(nouns);
        auto verb1 = pick(verbs);
        auto adv1 = pick(adverbs);
        auto adj2 = pick(adjectives);
        auto noun2 = pick(nouns);
        auto prep1 = pick(prepositions);
        auto place1 = pick(places);
        
        result = "The " + adj1 + " " + noun1 + " " + verb1 + " " + adv1 + ",\n";
        result += "While " + adj2 + " " + noun2 + "s rest " + prep1 + " " + place1 + ".\n";
        result += "With " + pick(adjectives) + " delight and " + pick(adjectives) + " glee,\n";
        result += "They " + pick(verbs) + " together wild and free!";
        
        return result;
    }
    
    std::string generateTemplate2() {
        return pick(adjectives) + " " + pick(nouns) + "\n" +
               pick(verbs) + " " + pick(adverbs) + " " + pick(prepositions) + " " + pick(places) + "\n" +
               pick(adjectives) + " " + pick(nouns) + " " + pick(verbs);
    }
    
    std::string generateTemplate3() {
        auto noun1 = pick(nouns);
        auto noun2 = pick(nouns);
        auto noun3 = pick(nouns);
        
        std::string result;
        result = "Once a " + pick(adjectives) + " " + noun1 + " met a " + pick(adjectives) + " " + noun2 + ",\n";
        result += "They " + pick(verbs) + " " + pick(adverbs) + " " + pick(prepositions) + " " + pick(places) + ".\n";
        result += "\"" + capitalize(pick(adjectives)) + " " + pick(nouns) + "!\" the " + noun1 + " cried,\n";
        result += "As a " + pick(adjectives) + " " + noun3 + " " + pick(verbs) + " by its side.";
        
        return result;
    }
    
    std::string generateTemplate4() {
        auto verb = pick(verbs);
        auto noun = pick(nouns);
        auto adj = pick(adjectives);
        auto adv = pick(adverbs);
        
        std::string result;
        result = "The " + noun + " " + verb + ",\n";
        result += adv + " it " + verb + ",\n";
        result += adj + " and " + pick(adjectives) + " it " + verb + ",\n";
        result += "Oh, how that " + noun + " " + verb + "!";
        
        return result;
    }
    
    std::string generateTemplate5() {
        auto noun1 = pick(nouns);
        auto noun2 = pick(nouns);
        
        std::string result;
        result = "What happens when a " + pick(adjectives) + " " + noun1 + "\n";
        result += "Meets a " + pick(adjectives) + " " + noun2 + " " + pick(prepositions) + " " + pick(places) + "?\n";
        result += "They " + pick(verbs) + " and " + pick(verbs) + " " + pick(adverbs) + ",\n";
        result += "Creating " + pick(adjectives) + " surprises!";
        
        return result;
    }
    
    std::string generateTemplate6() {
        std::string result = "I saw:\n";
        result += "A " + pick(adjectives) + " " + pick(nouns) + " that " + pick(verbs) + " " + pick(adverbs) + ",\n";
        result += "A " + pick(adjectives) + " " + pick(nouns) + " " + pick(prepositions) + " " + pick(places) + ",\n";
        result += "And a " + pick(adjectives) + " " + pick(nouns) + " doing " + pick(adjectives) + " things,\n";
        result += "All while the " + pick(adjectives) + " " + pick(nouns) + " " + pick(verbs) + "!";
        
        return result;
    }

public:
    PoemGenerator() : rng(std::random_device{}()) {}
    
    std::string generatePoem() {
        std::vector<std::function<std::string()>> templates = {
            [this]() { return generateTemplate1(); },
            [this]() { return generateTemplate2(); },
            [this]() { return generateTemplate3(); },
            [this]() { return generateTemplate4(); },
            [this]() { return generateTemplate5(); },
            [this]() { return generateTemplate6(); }
        };
        
        return pick(templates)();
    }
    
    void generatePoems(int count) {
        std::cout << std::string(60, '=') << "\n";
        std::cout << "  🎭 NONSENSE POEM GENERATOR (C++) 🎭\n";
        std::cout << std::string(60, '=') << "\n\n";
        
        for (int i = 1; i <= count; i++) {
            std::cout << "Poem #" << i << ":\n";
            std::cout << std::string(60, '-') << "\n";
            std::cout << generatePoem() << "\n\n";
            
            if (i < count) {
                std::cout << "\n";
            }
        }
        
        std::cout << std::string(60, '=') << "\n";
    }
};

int main(int argc, char* argv[]) {
    int count = 3;
    
    if (argc > 1) {
        count = std::atoi(argv[1]);
        if (count < 1) {
            std::cerr << "Usage: " << argv[0] << " [number_of_poems]\n";
            std::cerr << "Example: " << argv[0] << " 5\n";
            return 1;
        }
    }
    
    PoemGenerator generator;
    generator.generatePoems(count);
    
    return 0;
}
